### PR TITLE
[ISSUE #4731] HttpRequestProcessor enhancement

### LINKED ISSUE #4731: [Enhancement] Optimize the HttpRequestProcessor can obtain executor
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/eventmesh/issues?q=is%3Aissue) and found no similar issues.

### Enhancement Request

Optimize the HttpRequestProcessor can obtain executor  

### Describe the solution you'd like

Optimize the HttpRequestProcessor can obtain executor  

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct) *

### PR DESCRIPTION
**Pull request by karsonto**
Fixes #4731.

### Motivation

*Explain the content here.*
*Explain why you want to make the changes and what problem you're trying to solve.*

### Modifications

*Describe the modifications you've done.*

### Documentation

- Does this pull request introduce a new feature? (yes / no)
- If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
- If a feature is not applicable for documentation, explain why?
- If a feature is not documented yet in this PR, please create a followup issue for adding the documentation

### CODE REVIEW

File: eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/core/protocol/http/processor/inf/HttpRequestProcessor.java [hunk 1]
```diff
@@ -67,4 +68,10 @@ default String getExtension(CloudEvent event, String protocolKey) {
         return Objects.isNull(extension) ? "" : extension.toString();
     }
 
+    /**
+     *
+     * @return {@link Executor}
+     */
+    Executor executor();
+
```
Comment (Pil0tXia) [inline eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/core/protocol/http/processor/inf/HttpRequestProcessor.java:76]: It seems this method has no functional relation with `HttpRequestProcessor` interface. As for the current implementation of EventMesh for asynchronous requests, the executor has not been placed within the interface that handles the requests. May you please explain its design?

**Review by mxsm — COMMENTED** [review]
@karsonto  Please fix ci error

**Comment by Alonexc** [comment]
Conflicts need to be resolved.

**Comment by karsonto** [comment]
> Conflicts need to be resolved.

Please help to review again

**Comment by xwm1992** [comment]
@karsonto please resolve the conflicts thanks.

**Comment by karsonto** [comment]
> @karsonto please resolve the conflicts thanks.

ok,please review.

**Review by xwm1992 — APPROVED** [review]

File: eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/core/protocol/http/processor/AbstractHttpRequestProcessor.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+ * limitations under the License.
+ */
+
+package org.apache.eventmesh.runtime.core.protocol.http.processor;
+
+import org.apache.eventmesh.runtime.core.protocol.http.processor.inf.HttpRequestProcessor;
+
+public abstract class AbstractHttpRequestProcessor implements HttpRequestProcessor {
+
+}
```
Comment (Pil0tXia) [inline eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/core/protocol/http/processor/AbstractHttpRequestProcessor.java:24]: An additional blank abstract layer was added here. Is there any further modification on `AbstractHttpRequestProcessor`?

Comment (Pil0tXia) [reply eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/core/protocol/http/processor/AbstractHttpRequestProcessor.java:24]: This abstract class may be used in the future.

**Pull request merged**